compiler/rustc_attr/src/builtin.rs

```diff
@@ -28,7 +28,7 @@ pub fn rust_version_symbol() -> Symbol {
 }
 
 pub fn is_builtin_attr(attr: &Attribute) -> bool {
-    attr.is_doc_comment() || attr.ident().filter(|ident| is_builtin_attr_name(ident.name)).is_some()
+    attr.is_doc_comment() || attr.ident().map_or(false, |ident| is_builtin_attr_name(ident.name))
```

Comment:
`Option::is_some_and` seems clearer and communicates the intent better?

Reply:
agreed! Thanks for the great suggestion!